var/spack/repos/builtin/packages/e4s-cl/package.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class E4sCl(PythonPackage):
+    """Container Launcher for E4S containers, facilitating MPI library
+    translations"""
+
+    maintainers = ["spoutn1k", "FrederickDeny"]
+    homepage = "https://e4s-cl.readthedocs.io"
+    url = "https://oaciss.uoregon.edu/e4s/e4s-cl/releases"
+    git = "https://github.com/E4S-Project/e4s-cl"
+
+    tags = ["e4s"]
+
+    version("master", branch="master")
+    version("1.0.1", commit="b2c92993e0c7cb42de07f0f7cc02da3a06816192")
+    version("1.0.0", commit="410bb2e6601d9b90243a487ad7f7d2dabd8ba04c")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-termcolor@1.1.0", type=("build", "run"))
+    depends_on("py-pyyaml@6.0:", type=("build", "run"))
+    depends_on("py-texttable@1.6.2:", type=("build", "run"))
+    depends_on("py-python-ptrace@0.9.7:", type=("build", "run"))
+    depends_on("py-pyelftools@0.27", type=("build", "run"))
+    depends_on("py-requests@2.26.0:", type=("build", "run"))
+    depends_on("py-tinydb@4.7.0", type=("build", "run"))
+    depends_on("py-python-sotools@0.1.0", type=("build", "run"))
```

Comment:
I would still add the docker dep, otherwise `pip check` will fail saying a required dep is missing

Reply:
Well this commit is about to exponentially grow in size and as I am sure you noticed, packaging is not my day job.

Reply:
It's not mine either 😄 

You could also ask the developers not to list docker as a required dep and patch the requirements file if that sounds easier.

Reply:
From what I gather, that would require creating a new version and removing the older ones ?

What are the consequences of a `pip check` ? That docker module also lacks a ton of dependencies anyways.

Reply:
You don't have to create a new version, just add a patch file: https://spack.readthedocs.io/en/latest/packaging_guide.html#patches. But I would encourage you to also submit this patch upstream so the developers can review it and tell us whether or not docker really is required.

No consequences of `pip check` at the moment. We currently use pip to build all Python packages, and tell it to ignore dependencies. But in the future, I'm planning on switching to build/installer, which have a way of checking to make sure deps are installed without actually downloading them, so this package will break when I do that.

Reply:
I am the developer :) This is an MPI utility, and docker will *not* work with most MPI. This dependency is here by popular demand but not useful in production, and is used by one of the modular backends of the tool.

I will try and update the `py-docker` package, and see how annoying it proves to be.

One of the dependencies is `py-enum34`, which on my local clone of `spack` is an empty directory. Is this normal ?

Reply:
That is a good question, I had not found this option when I was looking for it. I applied a patch for the two versions defined and the next one will have `docker` in the extras.